float8_experimental/float8_linear.py

```diff
@@ -71,6 +71,54 @@ def _maybe_initialize_amaxes_scales_for_float8_cast(
         scale.copy_(new_scale)
 
 
+# this code was resurrected from https://github.com/pytorch-labs/float8_experimental/pull/128/files
+@torch._dynamo.allow_in_graph
+class manual_float8_mm(torch.autograd.Function):
+    """
+    Like torch.mm, but with X and W in float8
+    """
+
+    @staticmethod
+    def forward(
+        ctx,
+        x_fp8,
+        w_fp8_t,
+    ):
+        ctx.save_for_backward(x_fp8, w_fp8_t)
+        orig_shape = x_fp8.shape
+        x_fp8_reshaped = x_fp8.reshape(-1, orig_shape[-1])
```

Comment:
I think a con of this approach is that we dont get to capitalize on the nn.linear decomp ( this worth it). But mabye just a small coment on what this reshape and is doing:
Same thing was needed here: https://github.com/pytorch-labs/float8_experimental/pull/305/files#diff-993c030e4146627dff9686ff1c980699f88ca7f6152a4e809019a4dda645e770R129-R132

maybe make view?